common/Resources/ti.internal/extensions/js/console.js

```diff
@@ -1,30 +1,160 @@
-import { formatWithOptions } from '../node/internal/util/inspect';
+import { formatWithOptions, inspect } from '../node/internal/util/inspect';
 
-const nativeDebug = console.debug;
-const nativeError = console.error;
-const nativeInfo = console.info;
-const nativeLog = console.log;
-const nativeWarn = console.warn;
+function logTime(self, label, logData) {
+	label = `${label}`;
+	const startTime = self._times.get(label);
+	if (!startTime) {
+		process.emitWarning(`Label "${label}" does not exist`);
+		return true;
+	}
+	const duration = Date.now() - startTime;
+	if (logData) {
+		self.log(`${label}: ${duration}ms`, ...logData);
+	} else {
+		self.log(`${label}: ${duration}ms`);
+	}
+	return false;
+}
 
 const kColorInspectOptions = { colors: true };
 const kNoColorInspectOptions = {};
 
-console.debug = function (...args) {
-	nativeDebug.call(console, formatWithOptions(kColorInspectOptions, ...args));
-};
+class Console {
+	constructor(options, stderr, ignoreErrors) {
+		if (options && options.apiName === 'Ti.API') {
+			// Passing in Ti.API module where we retain log levels
+			this._apiModule = options;
+		} else {
+			// Node.JS streams
+			if (!options || typeof options.write === 'function') {
+				// no args, or first arg is a stream
+				options = {
+					stdout: options,
+					stderr,
+					ignoreErrors
+				};
+			}
+			this._stdout = options.stdout; // TODO: enforce has write function?
+			this._stderr = options.stderr || this._stdout;
+			this._ignoreErrors = options.ignoreErrors || true;
+			this._colorMode = options.colorMode || 'auto'; // TODO: enforce boolean or 'auto'
+			this._inspectOptions = options.inspectOptions; // TODO: enforce undefined or typeof 'object'
+		}
 
-console.error = function (...args) {
-	nativeError.call(console, formatWithOptions(kNoColorInspectOptions, ...args));
-};
+		this._times = new Map();
+		this._counts = new Map();
+		this._groupIndent = '';
+	}
 
-console.info = function (...args) {
-	nativeInfo.call(console, formatWithOptions(kColorInspectOptions, ...args));
-};
+	_writeToConsole(level, string) {
+		if (this._groupIndent.length !== 0) {
+			if (string.includes('\n')) {
+				string = string.replace(/\n/g, `\n${this._groupIndent}`);
+			}
+			string = this._groupIndent + string;
+		}
 
-console.log = function (...args) {
-	nativeLog.call(console, formatWithOptions(kColorInspectOptions, ...args));
-};
+		// Support wrapping Ti.API (which retains log level)
+		if (this._apiModule) {
+			this._apiModule[level](string);
+		} else {
+			// Support Node.JS streams like stdout/stderr which don't have log levels
+			const stream = (level === 'warn' || level === 'error') ? this._stderr : this._stdout;
+			// TODO: Handle this._ignoreErrors by doing try/catch/finally, hanging error handlers
```

Comment:
This sounds rather trivial and i think should be added since the default value for `ignoreErrors` is `true`.